providers/inwx/inwxProvider.go

```diff
@@ -0,0 +1,292 @@
+package inwx
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+
+	"github.com/svenpeter42/goinwx"
+)
+
+/*
+INWX Registrar and DNS provider
+
+Info required in `creds.json`:
+	- username
+	- password
+	- totp (TOPT code if 2FA is enabled)
+
+Additional settings available in `creds.json`:
+	- sandbox (set to 1 to use the sandbox API from INWX)
+
+*/
+
+type InwxApi struct {
+	client  *goinwx.Client
+	sandbox bool
+}
+
+var InwxDefaultNs = []string{"ns.inwx.de", "ns2.inwx.de", "ns3.inwx.eu", "ns4.inwx.com", "ns5.inwx.net"}
+var InwxSandboxDefaultNs = []string{"ns.ote.inwx.de", "ns2.ote.inwx.de"}
+
+var features = providers.DocumentationNotes{
+	providers.CanUseAlias:            providers.Cannot("INWX does not support the ALIAS or ANAME record type."),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUseDS:               providers.Unimplemented("DS records require a different API call that hasn't been implemented yet."),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseNAPTR:            providers.Can(),
+	providers.CanUseSRV:              providers.Can("SRV records with empty targets are not supported."),
+	providers.CanUseSSHFP:            providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.CanUseTXTMulti:         providers.Cannot("INWX only supports a single entry for TXT records"),
+	providers.CanAutoDNSSEC:          providers.Unimplemented("Supported by INWX but not implemented yet."),
+	providers.DocOfficiallySupported: providers.Cannot(),
+	providers.DocDualHost:            providers.Can(),
+	providers.DocCreateDomains:       providers.Unimplemented("Supported by INWX but not implemented yet."),
+	providers.CanGetZones:            providers.Can(),
+	providers.CanUseAzureAlias:       providers.Cannot(),
+}
+
+func init() {
+	providers.RegisterRegistrarType("INWX", newInwxReg)
+	providers.RegisterDomainServiceProviderType("INWX", newInwxDsp, features)
+}
+
+func newInwx(m map[string]string) (*InwxApi, error) {
+	for key := range m {
+		switch key {
+		case "username",
+			"password",
+			"totp",
+			"sandbox",
+			"domain":
+			continue
+		default:
+			return nil, fmt.Errorf("INWX: unknown key in `creds.json` (%s)\n", key)
+		}
+	}
+
+	if m["username"] == "" {
+		return nil, fmt.Errorf("INWX Username must be provided.")
+	}
+	if m["password"] == "" {
+		return nil, fmt.Errorf("INWX Password must be provided.")
+	}
+
+	var sandbox bool
+	if m["sandbox"] == "1" {
```

Comment:
Suggestion: `sandbox = m["sandbox"] == "1"`

Reply:
nice, that makes the code much easier to read!